### DISCUSSION TITLE
Turn by turn navigation camera logic.

### DISCUSSION THREAD

**Discussion by vicmpen** [react-native-maps/react-native-maps#5013] [Q&A]
Hello! 

So, i am using the package in order to render a map. I get routing information from a 3rd party service and i draw a polyline on the MapView. Now, when the user wished to, she can navigate from one place to another. Using a 3rd party service, i get the routing information with waypoints, bearing etc. 

My question is: is there a specific react-native-maps functionality that can assist with the turn-by-turn navigation feature, or i have to manually keep track of the location, the polyline length and the camera bearing/pitch myself?

Thanks in advance!

**Comment by aliyousafzai183** [comment]
Were you able to get any good package or you implemented everything yourself?

**Comment by vicmpen** [comment]
Did implement it myself. 

It’s doable, it just needs a lot of fine tuning here and there, in order to get the camera to behave exactly how you want it. 

Interesting challenge overall!

**Reply by aliyousafzai183** [reply]
Can you please share the code or references on how did you implemented it?
my email is warisallah923@gmail.com

And another question. Why you didn`t worked on implemented turn by turn navigation SDK via native modules?
